packages/node/src/transports/new.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  BaseTransportOptions,
+  createTransport,
+  NewTransport,
+  TransportMakeRequestResponse,
+  TransportRequest,
+} from '@sentry/core';
+import { TransportRequestExecutor } from '@sentry/core/dist/transports/base';
+import { eventStatusFromHttpCode } from '@sentry/utils';
+import * as fs from 'fs';
+import * as http from 'http';
+import * as https from 'https';
+
+import { HTTPModule, HTTPModuleClientRequest } from './base/http-module';
+
+interface HttpTransportOptions extends BaseTransportOptions {
+  // Todo: doc
+  headers?: Record<string, string>;
+  // TODO: doc Set a HTTP proxy that should be used for outbound requests.
+  proxy?: string;
+  // TODO: doc HTTPS proxy certificates path
+  caCerts?: string;
+  // Todo: doc
+  httpModule?: HTTPModule;
+}
+
+// TODO(v7):
+// - Rename this file "transports.ts"
+// - Move this file one folder upwards
+// - Delete "transports" folder
+
+/**
+ * TODO Doc
+ */
+export function makeNewHttpTransport(options: HttpTransportOptions): NewTransport {
+  // Proxy prioritization: http  => `options.proxy` | `process.env.http_proxy`
+  const proxy = filterNoProxy(options.url, options.proxy || process.env.http_proxy);
+
+  const httpModule = options.httpModule ?? http;
+
+  const agent = proxy
+    ? (new (require('https-proxy-agent'))(proxy) as http.Agent)
+    : new http.Agent({ keepAlive: false, maxSockets: 30, timeout: 2000 });
+
+  const requestExecutor = createRequestExecutor(options, httpModule, agent);
+  return createTransport({ bufferSize: options.bufferSize }, requestExecutor);
+}
+
+/**
+ * TODO Doc
+ */
+export function makeNewHttpsTransport(options: HttpTransportOptions): NewTransport {
```

Comment:
There's still a lot of duplicate code between `makeNewHttpTransport` and `makeNewHttpsTransport`, I wonder if we can better consolidate the code between them.

I would be open to perhaps making a uniform transport like what was discussed in https://github.com/getsentry/sentry-javascript/issues/2549, and make `http` vs `https` an option.

Reply:
> There's still a lot of duplicate code between `makeNewHttpTransport` and `makeNewHttpsTransport`, I wonder if we can better consolidate the code between them.

I believe it just *looks* like there is a lot of duplicate code because at a glance, the two functions look visually very similar. I don't think I would be able to consolidating the code any more without making stuff unnecessarily complicated. If you have concrete suggestions let me know!

> I would be open to perhaps making a uniform transport like what was discussed in #2549, and make `http` vs `https` an option.

I want to discuss this with the team to get som more opinions on this.